channeldb/migration32/migration_test.go

```diff
@@ -24,174 +25,208 @@ var (
-						OutgoingTimeLock: 4,
-						AmtToForward:     4,
-						BlindingPoint:    pubkey,
-						EncryptedData: []byte{
-							1, 2, 3,
-						},
-						TotalAmtMsat: 600,
-					},
-					// A hop with a blinding key and custom
-					// records.
-					{
-						PubKeyBytes:      testPub,
-						ChannelID:        800,
-						OutgoingTimeLock: 4,
-						AmtToForward:     4,
-						CustomRecords:    customRecord,
-						BlindingPoint:    pubkey,
-						EncryptedData: []byte{
-							1, 2, 3,
-						},
-						TotalAmtMsat: 600,
-					},
-				},
-			},
-		},
-		{
-			paymentResultCommon: paymentResultCommon2,
-			route: &Route{
-				TotalTimeLock: 101,
-				TotalAmount:   401,
-				SourcePubKey:  testPub2,
-				Hops: []*Hop{
-					{
-						PubKeyBytes:      testPub,
-						ChannelID:        800,
-						OutgoingTimeLock: 4,
-						AmtToForward:     4,
-						BlindingPoint:    pubkey,
-						EncryptedData: []byte{
-							1, 2, 3,
-						},
-						TotalAmtMsat: 600,
-					},
+		route: &Route{
+			TotalTimeLock: 101,
+			TotalAmount:   401,
+			SourcePubKey:  testPub2,
+			Hops: []*Hop{
+				{
+					PubKeyBytes:      testPub,
+					ChannelID:        800,
+					OutgoingTimeLock: 4,
+					AmtToForward:     4,
+					BlindingPoint:    pubkey,
+					EncryptedData:    []byte{1, 2, 3},
+					CustomRecords:    customRecord,
+					TotalAmtMsat:     600,
 				},
 			},
 		},
 	}
 
-	expectedResultsNew := []*paymentResultNew{
-		{
-			paymentResultCommon: paymentResultCommon1,
-			route: &mcRoute{
-				sourcePubKey: testPub,
-				totalAmount:  400,
-				hops: []*mcHop{
-					{
-						channelID:        100,
-						pubKeyBytes:      testPub,
-						amtToFwd:         500,
-						hasCustomRecords: true,
-					},
-					{
-						channelID:   800,
-						pubKeyBytes: testPub,
-						amtToFwd:    4,
-					},
-					{
-						channelID:        800,
-						pubKeyBytes:      testPub,
-						amtToFwd:         4,
-						hasBlindingPoint: true,
-					},
-					{
-						channelID:        800,
-						pubKeyBytes:      testPub,
-						amtToFwd:         4,
-						hasBlindingPoint: true,
-						hasCustomRecords: true,
-					},
+	//nolint:lll
+	resultNew1 = paymentResultNew{
+		id: 0,
+		timeFwd: tlv.NewPrimitiveRecord[tlv.TlvType1, uint64](
+			uint64(time.Unix(0, 1).UnixNano()),
+		),
+		timeReply: tlv.NewPrimitiveRecord[tlv.TlvType2, uint64](
+			uint64(time.Unix(0, 2).UnixNano()),
+		),
+		failureSourceIdx: tlv.SomeRecordT[tlv.TlvType5, uint8](
+			tlv.NewPrimitiveRecord[tlv.TlvType5, uint8](
+				uint8(failureIndex),
+			),
+		),
+		failure: tlv.SomeRecordT[tlv.TlvType6, failureMessage](
+			tlv.NewRecordT[tlv.TlvType6, failureMessage](
+				failureMessage{&lnwire.FailFeeInsufficient{}},
+			),
+		),
+		route: tlv.NewRecordT[tlv.TlvType3, mcRoute](mcRoute{
+			sourcePubKey: tlv.NewRecordT[tlv.TlvType0, Vertex](
+				testPub,
+			),
+			totalAmount: tlv.NewRecordT[tlv.TlvType1, lnwire.MilliSatoshi](400),
+			hops: tlv.NewRecordT[tlv.TlvType2, mcHops](mcHops{
```

Comment:
nit: could define test hops as their own top-level variable to avoid some of the indentation. Would probably also be a bit easier to review.

Reply:
good idea - updated 👍